lm-simulator/job/application.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import random
+import socket
+import sys
+from time import sleep
+import urllib.error
+import urllib.parse
+import urllib.request
+
+# Modify this value to reflect the IP address and port that the
+# license-manager-simulator is listening on in your environment.
+URL = "http://localhost:8000"
+
+
+def make_request(url, method="GET", data=None, headers=None):
+    """Make an HTTP request and return the response."""
+    if data is not None:
+        data = json.dumps(data).encode("utf-8")
+
+    if headers is None:
+        headers = {"Content-Type": "application/json"}
+
+    request = urllib.request.Request(url=url, data=data, headers=headers, method=method)
+
+    try:
+        with urllib.request.urlopen(request) as response:
+            return response.getcode(), response.read().decode("utf-8")
+    except urllib.error.HTTPError as e:
+        print(f"HTTP Error: {e.code} - {e.reason}")
+        sys.exit(1)
+    except urllib.error.URLError as e:
+        print(f"URL Error: {e.reason}")
+        sys.exit(1)
+
```

Comment:
Seems like there should be a final catch-all `except` here in case there are any other exceptions raised by `urllib`. Maybe something like: 
```suggestion
except Exception as e:
    print(f"Unknown Error: {e}")
    sys.exit(1)
```